compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -45,8 +45,17 @@ pub trait Delegate<'tcx> {
         place_with_id: &PlaceWithHirId<'tcx>,
         diag_expr_id: hir::HirId,
         bk: ty::BorrowKind,
+        is_autoref: bool,
     );
 
+    /// The value found at `place` is being copied.
+    /// `diag_expr_id` is the id used for diagnostics (see `consume` for more details).
+    fn copy(&mut self, place_with_id: &PlaceWithHirId<'tcx>, diag_expr_id: hir::HirId) {
+        // In most cases, treating a copy as a borrow is the right thing, so we forward
+        // this to the borrow callback by default.
```

Comment:
Let's explain why --

"In most cases, copying data from `x` is equivalent to doing `*&x`, so treat this as a borrow of `x`."

Reply:
Done